storage/innobase/buf/buf0flu.cc

```diff
@@ -2347,9 +2364,12 @@ static void buf_flush_page_cleaner()
       }
       while (false);
 
+      mysql_mutex_lock(&buf_pool.flush_list_mutex);
       if (!buf_pool.need_LRU_eviction())
```

Comment:
This check actually depends on `buf_pool.mutex`, not `buf_pool.flush_list_mutex`. Maybe we should remove the ThreadSanitizer suppression and add `mysql_mutex_assert_owner(&mutex)` to `buf_pool_t::need_LRU_eviction()`? It would seem to be the proper thing to do with respect to the bug scenario that we are attempting to fix. Note that the `buf_pool.mutex` would have to be acquired before `buf_pool.flush_list_mutex`. But, we should also keep in mind that the result of this check would be stale as soon as we no longer hold `buf_pool.mutex`.

Reply:
yes, thanks for pointing it out. Surprisingly we never acquire buf_pool.mutex before calling need_LRU_eviction(). May be performance reasons as the atomic variable try_LRU_scan would always be correct eventually. I added buf_pool.flush_list_mutex to make it similar to all other calls in the function but I think is not necessary. I have removed this change.